src/executor/Notifiable.hxx

```diff
@@ -46,7 +46,7 @@ class Notifiable : public Destructable
 public:
     /// Generic callback.
     virtual void notify() = 0;
-#ifdef __FreeRTOS__
+#if OPENMRN_FEATURE_MUTEX_FREERTOS
```

Comment:
I'm thinking whether this is the best feature macro to use here, vs. adding a new feature test macro.
Ultimately _from_isr() would need to exist for any RTOS, including an RTOS that has say pthreads APIs.

Reply:
I wasn't sure either and considered adding a new define but all usages of this appeared to be on OSSem.notify_from_isr which used this define.

Reply:
let's add a new feature macro OPENMRN_FEATURE_NOTIFY_FROM_ISR

Reply:
I'll update the refs that are in this PR to this, I spotted a few others that might make sense but will hold them back for a diff PR since I'm not certain of their usages.